### DISCUSSION TITLE
dynamic routing

### DISCUSSION THREAD

**Discussion by rufataliy** [vercel/next.js#19411] [Help]
Hey guys

I am trying to implement routing where when link is clicked sidebar appears displaying requested content while the browser's location changes as well but without rendering the main page.

You can see an example [here](https://recippy.rufat.tech/). So I wanna achieve the same functionality with nextjs. The one in the link is CRA created version.

Thanks

**Comment by banerjeesouvik** [comment]
I think you can handle this using next shallow routing.

**Reply by rufataliy** [reply]
Well, I can't because I change the URL from "/' to "/recipes/:id"

**Comment by Tomburgs** [comment]
It's the same page, just different states.

You want to create two pages:
pages/recipe/[recipeId].tsx which does something like this
```js
// Return same code as on home page.
export { default } from '../index.tsx';
```

and do the main logic in the pages/index.tsx file, something like so:
```jsx
import { useRouter } from 'next/router';

export default function Home() {
  // ...Product fetching goes here...

  // Get the specified recipeId from the url.
  const { query: { recipeId } } = useRouter();

  // ...Finding selected product logic goes here...

  // Return the JSX here, show the sidebar depending on if  the `recipeId` parameter is not `undefined`
}
